### ISSUE TITLE
Alacritty crashes when printing errors without stderr available

### ISSUE THREAD

**Issue by expectocode** [jwilm/alacritty#1457]
Which operating system does the issue occur on?
Fedora 28
If on linux, are you using X11 or Wayland?
X11, bspwm with sxhkd

I initially encountered this bug when launching alacritty with a `sxhkd` binding. Usually, alacritty would behave as expected, but following a strange systemd-related event where messages like this would be generated,
```Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 226:0
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:67
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:71
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:73
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:72
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:70
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:68
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:74
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:65
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:76
Jul 21 14:36:27 f /usr/libexec/gdm-x-session[2021]: (II) systemd-logind: got pause for 13:66
```

I would get (after restarting sxhkd to regain keyboard input) this bug in alacritty. Gnome-terminal (launched from alacritty) would open nvim fine. Also, running alacritty from alacritty would allow the spawned alacritty to open nvim, and it would print "Unhandled CSI" errors into the terminal used to spawn it. (https://github.com/jwilm/alacritty/issues/857#issuecomment-405226776). 

Finally, I tried changing my `sxhkd` binding from `alacritty` to `alacritty > /dev/null`. This fixed the problem, and leads me to suspect that alacritty will crash if it needs to output these "unhandled CSI" errors without a stdout to print to. 

It is, however, also possible that this is an sxhkd bug in spawning alacritty. And there is certainly some problem with systemd-logind creating these pause messages during normal usage (not at lid close time). Sorry for the big mess, hopefully this is not too complicated to fix.

**Comment by chrisduerr** [comment]
Hmm, your guess that Alacritty will crash when it doesn't have an stdout to print to doesn't seem completely unlikely, however for debugging it would be nice to have a backtrace.

This command will provide a full backtrace while still printing to stdout:
`RUST_BACKTRACE=1 ./target/debug/testing 2>&1 | tee /tmp/alacritty.backtrace`

If you can get this to work (it might need some tweaking to be used with `sxhkd`), plase share the full backtrace so the exact error can be pinpointed more easily.

**Comment by chrisduerr** [comment]
Thanks for that backtrace, would you mind trying the PR I've opened [here](1459)? Hopefull that should fix this issue.

**Comment by expectocode** [comment]
That seems to have fixed it! I guess we can close this issue when the PR is merged?

**Comment by chrisduerr** [comment]
Thanks for the feedback! The issue will be automatically closed once the PR is merged.

**Comment by expectocode** [comment]
Wait, i may have spoken too soon. sxhkd is being weird

**Comment by chrisduerr** [comment]
If you still get crashes, please share the backtrace and I'll get on it.

**Comment by expectocode** [comment]
With your PR branch, `alacritty` fails to launch unless stderr is redirected somewhere. I suspect I might not have noticed, had it not been for the config file errors from my scrollback config. I can't give you a traceback because stderr being redirected makes the bug disappear.

**Comment by chrisduerr** [comment]
This is why I've tried using `tee` to redirect output and still print it.
`RUST_BACKTRACE=1 ./target/debug/testing |& tee /tmp/alacritty.backtrace`

This is pretty much the same command I've printed earlier, but I think there was a misconception. This does not redirect `stdout` but rather split `stderr` to print both to the file and to stdout.

You could also try the following:
`RUST_BACKTRACE=1 ./target/debug/testing 2>&1 tee /tmp/alacritty.backtrace 1>&2`
This will redirect stderr to stdout, then split it with `tee` and after splitting it to a file it will redirect the stderr back to stdout.

**Comment by expectocode** [comment]
Pending our coredump generation, I found one place which emits stderr messages and appears to cause this bug (assuming my "stderr has nowhere to output and crashes the program" theory): https://github.com/jwilm/alacritty/blob/dbcb5885adb1f0b23c29838ef8dd837212322409/src/config.rs#L73

I tracked this down since I'm getting these config errors from using a scrollback config with your fix branch (which is a branch off master and hence doesn't have scrollback), and my sxhkd binded alacritty fails to launch if stderr is not redirected, but if i redirect it with /path/to/your/alacritty 2> /tmp/logfile then these config errors appear in the stderr file.

https://doc.rust-lang.org/std/macro.eprintln.html#panics says that eprintln! will panic if it can't output to stderr.

https://github.com/jwilm/alacritty/search?q=eprintln%21 shows that there are 4 uses of this macro in alacritty. If these had some error handling (or ignoring) then this specific class of crashes should go away.

**Comment by expectocode** [comment]
Related, https://github.com/rust-lang/rust/issues/24821 has the language discussion where the maintainers seem to suggest that mature programs should just eventually ignore the error.

https://github.com/vincentdephily/emlop/pull/4/files is an example of a way to fix this, by using .unwrap_or(()) to suppress the panic.
